Framework/DataHandling/src/SaveNexusGeometry.cpp

```diff
@@ -0,0 +1,100 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+/* SaveNexusGeometry : A thin Algorithm wrapper over
+ * NexusGeometry::saveInstrument allowing user to save the geometry from
+ * instrument attached to a workspace.
+ *
+ * @author Takudzwa Makoni, RAL (UKRI), ISIS
+ * @date 16/08/2019
+ */
+#include "MantidDataHandling/SaveNexusGeometry.h"
+#include "MantidAPI/ExperimentInfo.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidGeometry/Instrument/InstrumentVisitor.h"
+#include "MantidNexusGeometry/NexusGeometrySave.h"
+
+#include <memory>
+#include <utility>
+
+namespace Mantid {
+namespace DataHandling {
+using Mantid::API::WorkspaceProperty;
+using Mantid::Kernel::Direction;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(SaveNexusGeometry)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+
+const std::string SaveNexusGeometry::name() const {
+  return "SaveNexusGeometry";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+
+int SaveNexusGeometry::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+
+const std::string SaveNexusGeometry::category() const {
+  return "DataHandling\\Instrument";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+
+const std::string SaveNexusGeometry::summary() const {
+  return "Save the instrument from a workspace to a Nexus-format"
+         " HDF file. WARNING: shapes are NOT saved in the present version "
+         "(1.0).";
+}
+
+//----------------------------------------------------------------------------------------------
+
+/** Initialize the algorithm's properties.
+ */
+void SaveNexusGeometry::init() {
+
+  // TODO resolve linkererror for experimentinfo, replace MatrixWorkspace with
+  // base class ExperimentInfo.
+  declareProperty(std::make_unique<WorkspaceProperty<API::MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "Workspace containing the Instrument.");
+
+  declareProperty(std::make_unique<API::FileProperty>(
+                      "Filename", "", API::FileProperty::OptionalSave),
+                  "Full path to save destination file");
+
+  declareProperty("H5Path", "entry" /*default*/,
```

Comment:
Understanding what this actually does now my suggestion for this name is a poor one.
This should be changed to something that implies it is a single group name, not a path.

Reply:
Of Dan's comments, all of which I think are valid - this is the one that will need to be corrected with the highest priority. I may well fix it as part of my work on the `SaveNexusESS` algorithm.